src/instr.rs

```diff
@@ -1,23 +1,76 @@
 use parser;
+use edit::EditMode;
+use edit::ViMode;
 
 pub enum Instr {
```

Comment:
To keep the interpretation logic simple, it might make sense to come up with different instruction sets per mode. But I'm happy to merge your changes with a joint instruction set and take it from there.